### ISSUE TITLE
Constant ERR_NETWORK_CHANGED in Chrome

### ISSUE THREAD

**Issue by leoluk** [tailscale/tailscale#8384]
### What is the issue?

While tailscaled is running, this happens about ~1 per minute in Google Chrome:

<img width="531" alt="image" src="https://github.com/tailscale/tailscale/assets/859697/ce718592-fc59-4775-9e71-c5b002e84582">

At the same time, these logs are printed by tailscaled:

```
tailscaled[2128]: monitor: RTM_NEWROUTE: src=, dst=fd47:[...]::/64, gw=, outif=0, table=254
tailscaled[2128]: monitor: RTM_DELROUTE: src=, dst=fd47:[...]::/64, gw=fe80::[...], outif=2, table=254
tailscaled[2128]: monitor: RTM_NEWROUTE: src=, dst=fd47:[...]:1::/64, gw=, outif=0, table=254
tailscaled[2128]: monitor: RTM_DELROUTE: src=, dst=fd47:[...]:1::/64, gw=fe80::[...], outif=2, table=254
tailscaled[2128]: monitor: RTM_NEWROUTE: src=, dst=fd47:[...]:1::/64, gw=, outif=0, table=254
```

Nothing else is logged. The timing always lines up perfectly.

Not the same issue as https://github.com/tailscale/tailscale/issues/3686.

### OS

Linux

### OS version

NixOS 23.05

### Tailscale version

1.42.0, go1.20.4

### Other software

Other potential culprits:

- NetworkManager
- Docker (rootless, so not messing with routes)
- avahi-daemon

### Bug report

BUG-fb69d753a490f319bf22be1615edc4fb27ff1cb985b852871e5d765e5d47edc6-20230619151814Z-50769c47410ba9cb

**Comment by kelivel** [comment]
@leoluk I'm taking a look into this. In the meantime, can you try upgrading to 1.50 and let me know if that shows any change in behavior?

**Comment by jontze** [comment]
> @leoluk I'm taking a look into this. In the meantime, can you try upgrading to 1.50 and let me know if that shows any change in behavior?

I assume that this issue is fixed with 1.50. I had it quite often and after I upgraded it never appeared again. 

I'm currently on:
```
1.50.1
  tailscale commit: f45c02bfcf5ee5790c3af278c9e974c9b9b0e771-dirty
  go version: go1.21.1
```

**Comment by kelivel** [comment]
Great, glad to hear it! I never managed to reproduce it, so definitely let us know if it happens again and we'll continue digging in.
